spec/db/models.rb

```diff
@@ -27,24 +27,45 @@ class Author < VitualTotalTestBase
   virtual_total :total_recently_published_books, :recently_published_books
   virtual_aggregate :sum_recently_published_books_rating, :recently_published_books, :sum, :rating
 
-  # virtual_total using a virtual_has_many
+  # virtual_total using a virtual_has_many with an array of associations
+  # LAZY: sorry for lack of creativity, this could have been done with a where
   def named_books
     # I didn't have the creativity needed to find a good ruby only check here
     books.select { |b| b.name }
   end
 
+  # attribute with uses that depends upon a virtual column in another model.
+  # this tests a uses with a virtual column, a hash uses.
+  # LAZY: sorry for lack of creativity, this could be done with pure sql, or a delegate, but aiming for those usecases
```

Comment:
Here too:  I would just comment that this is a "poorly optimized use case" that we support.